inquire/src/prompts/path_select/modes.rs

```diff
@@ -0,0 +1,50 @@
+//! Prompt modes 
+//! 
+//! Everything here should be exposed to the (library) user
+use strum::{Display, EnumCount, FromRepr};
+
+
+/// Different path selection modes specify what the user can choose
+#[derive(Clone, Eq, PartialEq)]
+pub enum PathSelectionMode<'a> {
+    /// The user may pick a directory.
+    Directory,
+    /// The user may pick a file with the given (optional) extension.
+    File(Option<&'a str>),
+    /// The user can set gitignore rules from a file 
+    /// The user may pick multiple paths
+    Multiple(Vec<PathSelectionMode<'a>>),
+}
```

Comment:
I feel like this can be made simpler for end-users. On a first glance, it's not quite intuitive how this selection mode behaves, with the recursive Multiple warranting more attention.

wdyt of:

accept enum, like JS's accept attribute: All, Only(String), Any(Vec<String>)

Then PathSelectionMode
- Directory
- File(Accept)
- FileOrDirectory(Accept)

I still don't fee this is the best solution, but it feels more intuitive

Reply:
Sure, I suppose it was redundant the way it was. I tried implementing an inner  enum (`PathFilter`) that can accept or deny based on file extensions, stems, and matching functions. I also added some documentation tests for the different modes and tried to tighten up the documentation.